pkg/flow/temporary.go

```diff
@@ -488,8 +504,8 @@ func (engine *engine) doKnativeHTTPRequest(ctx context.Context,
 	engine.sugar.Debugf("function request for image %s name %s addr %v:", ar.Container.Image, ar.Container.ID, addr)
 	engine.logger.Debugf(ctx, engine.flow.ID, engine.flow.GetAttributes(), "function request for image %s name %s", ar.Container.Image, ar.Container.ID)
 
-	deadline := time.Now().UTC().Add(time.Duration(ar.Workflow.Timeout) * time.Second)
-	rctx, cancel := context.WithDeadline(context.Background(), deadline)
+	deadline := time.Now().UTC().Add(time.Duration(ar.fnContext.Timeout) * time.Second)
+	rctx, cancel := context.WithDeadline(ctx, deadline)
```

Comment:
Revert this line to `context.Background`. This is intentionally not the normal context because of the complex way we want functions to gracefully handle a cancellation.